http3/settings.go

```diff
@@ -0,0 +1,126 @@
+package http3
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/quicvarint"
+)
+
+const (
+	SettingQPACKMaxTableCapacity SettingID = 0x01
+	SettingMaxFieldSectionSize   SettingID = 0x06
+	SettingQPACKBlockedStreams   SettingID = 0x07
+
+	// https://www.ietf.org/archive/id/draft-ietf-masque-h3-datagram-02.html#name-http-settings-parameter
+	SettingDatagram SettingID = 0xffd276
+
+	// https://datatracker.ietf.org/doc/draft-ietf-masque-h3-datagram/00/
+	SettingDatagramDraft00 SettingID = 0x276
```

Comment:
Do we need backwards compatibility here? IMO it would be sufficient to support one draft version.

Reply:
Chromium via quiche, still uses `0x276`: https://github.com/google/quiche/blob/2682b515465886008918b88d0c3dee48414caa4f/quic/core/http/http_constants.h#L42

```cpp
  // draft-ietf-masque-h3-datagram.
  SETTINGS_H3_DATAGRAM = 0x276,
```